src/courtroom.cpp

```diff
@@ -1786,89 +1813,51 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
-  // Remove undesired newline chars
-  m_chatmessage[MESSAGE].remove("\n");
-  chatmessage_is_empty =
-      m_chatmessage[MESSAGE] == " " || m_chatmessage[MESSAGE] == "";
+  // If chatblank is enabled, use the character's name for logs
+  if (f_displayname.trimmed().isEmpty())
+    f_displayname = ao_app->get_showname(char_list.at(f_char_id).name);
 
-  if (f_char_id >= 0 && !chatmessage_is_empty &&
-      f_message == previous_ic_message) // Not a system message
-    return;
-
-  if (f_char_id <= -1)
-    previous_ic_message =
-        ""; // System messages don't care about repeating themselves
-  else
-    previous_ic_message = f_message;
-  bool ok;
-  int objection_mod = m_chatmessage[OBJECTION_MOD].toInt(
-      &ok, 10); // checks if its a custom obj.
+  // Check if a custom objection is in use
+  int objection_mod = 0;
   QString custom_objection = "";
-  if (!ok && m_chatmessage[OBJECTION_MOD].contains("4&")) {
+  if (m_chatmessage[OBJECTION_MOD].contains("4&")) {
     objection_mod = 4;
-    custom_objection = m_chatmessage[OBJECTION_MOD].split(
-        "4&")[1]; // takes the name of custom objection.
+    custom_objection = m_chatmessage[OBJECTION_MOD].split("4&")[1]; // takes the name of custom objection.
+  }
+  else{
+      objection_mod = m_chatmessage[OBJECTION_MOD].toInt();
   }
-  // Stop the chat arrow from animating
-  ui_vp_chat_arrow->stop();
 
-  text_state = 0;
-  anim_state = 0;
-  ui_vp_objection->stop();
-  chat_tick_timer->stop();
-  ui_vp_evidence_display->reset();
-
-  // Hey, our message showed up! Cool!
-  if (m_chatmessage[MESSAGE] == ui_ic_chat_message->text().remove("\n") &&
-      m_chatmessage[CHAR_ID].toInt() == m_cid) {
-    ui_ic_chat_message->clear();
-    if (ui_additive->isChecked())
-      ui_ic_chat_message->insert(" ");
-    objection_state = 0;
-    realization_state = 0;
-    screenshake_state = 0;
-    is_presenting_evidence = false;
-    if (!ao_app->is_stickypres_enabled())
-      ui_pre->setChecked(false);
-    ui_hold_it->set_image("holdit");
-    ui_objection->set_image("objection");
-    ui_take_that->set_image("takethat");
-    ui_custom_objection->set_image("custom");
-    ui_realization->set_image("realization");
-    ui_screenshake->set_image("screenshake");
-    ui_evidence_present->set_image("present");
+  // Reset IC display
+  reset_ic();
+
+  // Reset UI elements after client message gets sent
+  if (m_chatmessage[CHAR_ID].toInt() == m_cid) {
+    reset_ui();
   }
 
   // Let the server handle actually checking if they're allowed to do this.
   is_additive = m_chatmessage[ADDITIVE].toInt() == 1;
 
-  QString f_charname = "";
-  if (f_char_id >= 0)
-    f_charname = ao_app->get_showname(char_list.at(f_char_id).name);
-
   chatlogpiece *temp =
-      new chatlogpiece(f_charname, f_showname, m_chatmessage[MESSAGE], false, m_chatmessage[TEXT_COLOR].toInt());
+      new chatlogpiece(f_displayname, f_displayname, m_chatmessage[MESSAGE], false, m_chatmessage[TEXT_COLOR].toInt());
```

Comment:
The purpose of this was to allow enabling and disabling shownames while preserving the log - in other words, when you toggle shownames when the log is populated, the names will actually change retroactively. This change breaks that feature.

Reply:
right, my bad. i saw that f_displayname (formerly showname) and f_charname were equal, but i forgot they were different in the showname codepath. will revert this change.